### PR TITLE
Misc fixes, depend on pytest for testing

### PR DESCRIPTION
**Pull request by skirpichev**
This is mostly a cleanup of legacy code from looking on coverage data for tests.

Importing pytest in tests may be a problem, as it introduce an optional dependency, which will be required for testing of installed package with runtests().  But I think it worth always using pytest, instead of homegrown test framework.

There are options.  We can just document this optional requirement.  Or, we can also change package to use setuptools, which allow specification for optional requirements (see diofant's setup.py file for example).

### CODE REVIEW

**Comment by skirpichev** [comment]
CI tests pass: https://travis-ci.org/fredrik-johansson/mpmath/builds/432564662

But somehow, there is no travis checks here, in the pull request.  @fredrik-johansson, perhaps you changed integration with travis-ci?

File: mpmath/tests/test_gammazeta.py [hunk 1]
```diff
@@ -508,20 +508,6 @@ def test_gamma_huge_5():
         "7571892628368354580620654233316841")
     mp.dps = 15
 
-def test_gamma_huge_6():
```
Comment (fredrik-johansson) [inline mpmath/tests/test_gammazeta.py]: Why was this removed?

File: doc/source/setup.txt [hunk 2]
```diff
@@ -99,13 +99,7 @@ Alternatively, they can be run from the ``tests`` directory via
...
     ``python runtests.py``
 
-The tests should finish in about a minute. If you have `psyco <http://psyco.sourceforge.net/>`_ installed, the tests can also be run with
-
-    ``python runtests.py -psyco``
-
-which will cut the running time in half.
-
-If any test fails, please send a detailed bug report to the `mpmath issue tracker <https://github.com/fredrik-johansson/mpmath/issues>`_. The tests can also be run with `py.test <http://pylib.org/>`_. This will sometimes generate more useful information in case of a failure.
+If any test fails, please send a detailed bug report to the `mpmath issue tracker <https://github.com/fredrik-johansson/mpmath/issues>`_. The tests can also be run with `py.test <https://pytest.org/>`_. This will sometimes generate more useful information in case of a failure.
```
Comment (fredrik-johansson) [inline doc/source/setup.txt]: The documentation does not seem to reflect pytest being a requirement.

**Comment by fredrik-johansson** [comment]
This generally looks like a nice cleanup.

I'm -0 on requiring pytest, but it's fine if it's correctly documented and convenient for users. Does anyone else have an opinion?

File: mpmath/tests/test_gammazeta.py (hunk 1 above)
Comment (skirpichev) [reply mpmath/tests/test_gammazeta.py]: The test doesn't look right.  I'm not sure about numerical values used, do you?  And it's disabled, anyway (early "return" statement).

An option may be make this xfailed test.  But only after we make sure that tested values are correct.

File: doc/source/setup.txt (hunk 2 above)
Comment (skirpichev) [reply doc/source/setup.txt]: Correct.  Shall I just mention this requirement in the docs or I should also declare dependencies with setuptools in the setup.py?

**Comment by skirpichev** [comment]
@fredrik-johansson, travis-ci checks also absent on other recent pull requests in your repository.  I hope, this is intentional.

**Comment by fredrik-johansson** [comment]
No, it's not intentional. I don't know what happened.

File: doc/source/setup.txt (hunk 2 above)
Comment (fredrik-johansson) [reply doc/source/setup.txt]: No strong opinion either way.

File: mpmath/tests/test_gammazeta.py (hunk 1 above)
Comment (fredrik-johansson) [reply mpmath/tests/test_gammazeta.py]: You're right, the real part is not right. May as well remove it.

**Comment by skirpichev** [comment]
> No, it's not intentional. I don't know what happened.

Please check that CI tests now are enabled for pull requests.  BTW, it seems some CI tests on the master branch now are failing, probably after merged PR.

**Comment by fredrik-johansson** [comment]
Weird, I don't see any option for this in the settings on Travis.

And the repo settings on GH tell me that "GitHub Services are being deprecated" but everything seems to be on.

**Comment by skirpichev** [comment]
> Weird, I don't see any option for this in the settings on Travis.

Hmm, AFAIR it's "build pushed pull requests".

Check also settings for branches in the github.  Maybe you have some branch protection rules for the master branch.

**Comment by fredrik-johansson** [comment]
"build pushed pull requests" is enabled.

There are no branch protection rules.

**Comment by skirpichev** [comment]
Hmm, check authorized apps on your account settings page.  I have "Travis CI for Open Source" here.

**Comment by fredrik-johansson** [comment]
Same here (under OAuth Apps).

**Comment by skirpichev** [comment]
Well, I have no ideas anymore)  Perhaps, you could try revoke these grants + push Off/On the repo in travis-ci.  Or just ask GH support.

File: doc/source/setup.txt (hunk 2 above)
Comment (skirpichev) [reply doc/source/setup.txt]: Ok, pytest dependency now documented.  I also suggest using ``py.test --pyargs mpmath`` for testing on live installation.  Probably, runtests() could be removed eventually.

Let me know if you are ok with this.

**Comment by skirpichev** [comment]
Great, status checks are working again!

**Comment by skirpichev** [comment]
Added trivial fix for #401 and couple issues.  Another variant of the fix is in #410.  @fredrik-johansson, please let me know which solution do you prefer (I can cherry-pick @warnerjon12's fix) and if there are any other required changes for this pull request.

**Comment by fredrik-johansson** [comment]
OK, it seems reasonable.
